.github/workflows/CI.yml

```
@@ -41,4 +41,7 @@ jobs:
                     exit 1
                fi
       - run: julia -e 'using Pkg ; Pkg.add(["Singular_jll", "CxxWrap", "CMake"])'
-      - run: julia build.jl
+      # TODO: install development version of Singular instead of master...
+      - run: julia -e 'using Pkg ; Pkg.add(PackageSpec(url="https://github.com/oscar-system/Singular.jl", rev="master"))'
```

The following should work for the branch lookup in the main repository.
(Checking for the branch in a corresponding fork might also be possible but more involved...)

```suggestion
      - name: "Look for Singular.jl branch"
        id: detectbranch
        # look for a branch with the same name for Singular.jl
        continue-on-error: true
        run: julia -e "using Pkg; Pkg.add(PackageSpec(url=\"https://github.com/oscar-system/Singular.jl\", rev=\"${GITHUB_HEAD_REF#refs/heads/}\"));"
      - name: "Test Singular.jl master"
        # use master otherwise
        if: steps.detectbranch.outcome == 'failure'
        run: julia -e "using Pkg; Pkg.add(PackageSpec(url=\"https://github.com/oscar-system/Singular.jl\", rev=\"master\"));"
```